db.go

```diff
@@ -476,6 +476,40 @@ func (db *DB) AscendKeys(pattern []byte, filterExpired bool, handleFn func(k []b
 	})
 }
 
+// AscendKeysRange calls handleFn for keys within a range in the db in ascending order.
+// Since our expiry time is stored in the value, if you want to filter expired keys,
+// you need to set parameter filterExpired to true. But the performance will be affected.
+// Because we need to read the value of each key to determine if it is expired.
+func (db *DB) AscendKeysRange(startKey, endKey, pattern []byte, filterExpired bool, handleFn func(k []byte) (bool, error)) {
+	db.mu.RLock()
+	defer db.mu.RUnlock()
+
+	var reg *regexp.Regexp
+	if len(pattern) > 0 {
+		reg = regexp.MustCompile(string(pattern))
+	}
+
+	db.index.AscendRange(startKey, endKey, func(key []byte, pos *wal.ChunkPosition) (bool, error) {
+		if reg == nil || reg.Match(key) {
+			var invalid bool
+			if filterExpired {
+				chunk, err := db.dataFiles.Read(pos)
+				if err != nil {
+					return false, err
+				}
+				if value := db.checkValue(chunk); value == nil {
+					invalid = true
+				}
+			}
+			if invalid {
+				return true, nil
+			}
+			return handleFn(key)
+		}
+		return true, nil
+	})
+}
```

Comment:
Error Handling Improvements:
- Replace regexp.MustCompile with regexp.Compile to allow callers to handle compilation errors

Code Structure Optimization:
- Remove unnecessary 'invalid' variable
- Reduce if nesting
- Add essential comments for clarity

Here are my proposed changes. I welcome any feedback or discussion if you have different opinions.

```suggestion
func (db *DB) AscendKeys(pattern []byte, filterExpired bool, handleFn func(k []byte) (bool, error)) error {
	db.mu.RLock()
	defer db.mu.RUnlock()

	var reg *regexp.Regexp
	if len(pattern) > 0 {
		var err error
		reg, err = regexp.Compile(string(pattern))
		if err != nil {
			return err
		}
	}

	db.index.Ascend(func(key []byte, pos *wal.ChunkPosition) (bool, error) {
		if reg != nil && !reg.Match(key) {
			return true, nil
		}
		if filterExpired {
			chunk, err := db.dataFiles.Read(pos)
			if err != nil {
				return false, err
			}
			if value := db.checkValue(chunk); value == nil {
				return true, nil
			}
		}
		return handleFn(key)
	})

	return nil
}
```

Reply:
If you agree with my changes, I can provide the corresponding unit tests.

Reply:
I think your suggestion is necessary and I have revised my PR